src/commands/synthetics/utils.ts

```diff
@@ -283,30 +291,40 @@ export const waitForResults = async (
       .catch(() => (isTunnelConnected = false))
   }
 
-  while (triggerResults.filter((tr) => !tr.result).length) {
+  const getTest = (id: string): Test => tests.find((t) => t.public_id === id)!
+
+  while (inProgressTriggers.size) {
     const pollingDuration = new Date().getTime() - pollingStartDate
 
     // Remove test which exceeded their pollingTimeout
-    for (const triggerResult of triggerResults.filter((tr) => !tr.result)) {
+    for (const triggerResult of inProgressTriggers) {
```

Comment:
Unrelated to this PR but just realised the order of events in this loop makes us wait 5s (`POLLING_INTERVAL`) for nothing, and probably mark some tests as timed out that are received just after 🤔 
```
+ compute pollingDuration
- fail timed out tests
- fail tunnel tests
- stop if max timeout reached  <- extra 5s unnecessary wait if we're not even fetching anything
- fetch results                <- might receive results of tests marked as timed out just before
- wait 5s
```
Should probably be:
```
- fetch results
+ compute pollingDuration
- fail timed out tests
- fail tunnel tests
- stop if max timeout reached
- wait 5s
```

Reply:
Good catch, I'll create a card for that.